### DISCUSSION TITLE
question

### DISCUSSION THREAD

**Discussion by Double-Click-ERP** [chatwoot/chatwoot#6440] [Self hosted] [answered]
hello ,

how can i apply my change in .vue file  (ex. comment some lines)

thank you

**Comment by Double-Click-ERP** [comment]
ex. after editing the file "chatwoot/chatwoot/app/javascript/dashboard/routes/dashboard/conversation/contact/ContactInfo.vue" and comment some lines

>        
       

and restart 

> sudo systemctl restart chatwoot.target

the commented delete icon is still appear

**Reply by lastwall** [reply]
You need to recompile assets too
use this :

```
rake assets:precompile RAILS_ENV=production

systemctl restart chatwoot.target
```

**Reply by Double-Click-ERP** [reply]
>rake assets:precompile RAILS_ENV=production --trace

>Warning: the running version of Bundler (2.2.33) is older than the version that created the lockfile (2.3.16). We suggest you to upgrade to the version that created the lockfile by running `gem install bundler:2.3.16`.
Following files may not be writable, so sudo is needed:
  /usr/local/rvm/gems/ruby-3.0.4
  /usr/local/rvm/gems/ruby-3.0.4/bin
  /usr/local/rvm/gems/ruby-3.0.4/bin
  /usr/local/rvm/gems/ruby-3.0.4/build_info
  /usr/local/rvm/gems/ruby-3.0.4/bundler
  /usr/local/rvm/gems/ruby-3.0.4/cache
  /usr/local/rvm/gems/ruby-3.0.4/doc
  /usr/local/rvm/gems/ruby-3.0.4/environment
  /usr/local/rvm/gems/ruby-3.0.4/extensions
  /usr/local/rvm/gems/ruby-3.0.4/gems
  /usr/local/rvm/gems/ruby-3.0.4/plugins
  /usr/local/rvm/gems/ruby-3.0.4/specifications
  /usr/local/rvm/gems/ruby-3.0.4/wrappers
** Invoke assets:precompile (first_time)
** Invoke assets:environment (first_time)
** Execute assets:environment
** Invoke environment (first_time)
** Execute environment
[02/14/23 09:39:28 +0200 chat-VirtualBox (5467)] INFO : No configuration file loaded, and no configuration found in ENV. For assistance configuring Scout, visit https://docs.scoutapm.com/#ruby-configuration-options
[02/14/23 09:39:28 +0200 chat-VirtualBox (5467)] INFO : Scout Agent [5.2.0] Initialized
[02/14/23 09:39:28 +0200 chat-VirtualBox (5467)] INFO : Not Loading Instruments
[02/14/23 09:39:28 +0200 chat-VirtualBox (5467)] INFO : Scout Agent [5.2.0] Installed
[02/14/23 09:39:28 +0200 chat-VirtualBox (5467)] INFO : Monitoring isn't enabled for the [production] environment.
[02/14/23 09:39:28 +0200 chat-VirtualBox (5467)] INFO : Deferring agent start. Standing by for first request
** Invoke yarn:install (first_time)
** Execute yarn:install
yarn install v1.22.19
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
Done in 1.28s.
** Execute assets:precompile
rake aborted!
Errno::EACCES: Permission denied @ apply2files - /home/chatwoot/chatwoot/tmp/cache/assets/sprockets/v4.0.0/Jv/JvQrF9P6bEHt3C23ZydxesTvTJ8_QpUPDm29TRIFPrA.cache
/usr/local/rvm/rubies/ruby-3.0.4/lib/ruby/3.0.0/fileutils.rb:1147:in `utime'
/usr/local/rvm/rubies/ruby-3.0.4/lib/ruby/3.0.0/fileutils.rb:1147:in `block in touch'
/usr/local/rvm/rubies/ruby-3.0.4/lib/ruby/3.0.0/fileutils.rb:1144:in `each'
/usr/local/rvm/rubies/ruby-3.0.4/lib/ruby/3.0.0/fileutils.rb:1144:in `touch'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/cache/file_store.rb:72:in `get'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/cache.rb:223:in `get'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/cache.rb:90:in `fetch'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/base.rb:71:in `file_digest'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/unloaded_asset.rb:105:in `dependency_history_key'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/loader.rb:324:in `fetch_asset_from_dependency_cache'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/loader.rb:43:in `load'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/cached_environment.rb:44:in `block in load'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/cached_environment.rb:44:in `fetch'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/cached_environment.rb:44:in `load'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/base.rb:81:in `find_asset'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/base.rb:88:in `find_all_linked_assets'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/manifest.rb:125:in `each'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/manifest.rb:125:in `to_a'
/usr/local/rvm/gems/ruby-3.0.4/gems/sprockets-4.1.1/lib/sprockets/manifest.rb:125:in `block (2 levels) in find'
/usr/local/rvm/gems/ruby-3.0.4/gems/concurrent-ruby-1.1.10/lib/concurrent-ruby/concurrent/executor/safe_task_executor.rb:24:in `block in execute'
/usr/local/rvm/gems/ruby-3.0.4/gems/concurrent-ruby-1.1.10/lib/concurrent-ruby/concurrent/synchronization/mutex_lockable_object.rb:47:in `block in synchronize'
/usr/local/rvm/gems/ruby-3.0.4/gems/concurrent-ruby-1.1.10/lib/concurrent-ruby/concurrent/synchronization/mutex_lockable_object.rb:47:in `synchronize'
/usr/local/rvm/gems/ruby-3.0.4/gems/concurrent-ruby-1.1.10/lib/concurrent-ruby/concurrent/synchronization/mutex_lockable_object.rb:47:in `synchronize'
/usr/local/rvm/gems/ruby-3.0.4/gems/concurrent-ruby-1.1.10/lib/concurrent-ruby/concurrent/executor/safe_task_executor.rb:22:in `execute'
/usr/local/rvm/gems/ruby-3.0.4/gems/concurrent-ruby-1.1.10/lib/concurrent-ruby/concurrent/promise.rb:564:in `block in realize'
/usr/local/rvm/gems/ruby-3.0.4/gems/concurrent-ruby-1.1.10/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb:352:in `run_task'
/usr/local/rvm/gems/ruby-3.0.4/gems/concurrent-ruby-1.1.10/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb:343:in `block (3 levels) in create_worker'
/usr/local/rvm/gems/ruby-3.0.4/gems/concurrent-ruby-1.1.10/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb:334:in `loop'
/usr/local/rvm/gems/ruby-3.0.4/gems/concurrent-ruby-1.1.10/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb:334:in `block (2 levels) in create_worker'
/usr/local/rvm/gems/ruby-3.0.4/gems/concurrent-ruby-1.1.10/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb:333:in `catch'
/usr/local/rvm/gems/ruby-3.0.4/gems/concurrent-ruby-1.1.10/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb:333:in `block in create_worker'
Tasks: TOP => assets:precompile

**Comment by lastwall** [comment, marked as answer]
did you log in with chatwoot user?
try these:

```
sudo -i -u chatwoot
cd chatwoot
rake assets:precompile RAILS_ENV=production
sudo systemctl restart chatwoot.target
```

**Reply by Double-Click-ERP** [reply]
Thank You
